lighthouse-core/report/html/renderer/dom.js

```diff
@@ -77,6 +77,21 @@ class DOM {
     return element;
   }
 
+  /**
+   * @param {Node} node
+   */
+  removeCommentNodes(node) {
+    for (const child of Array.from(node.childNodes)) {
```

Comment:
wdyt about using the TreeWalker API?

https://stackoverflow.com/a/16152333

I just tried it out and this current implementation takes ~40ms for an entire report DOM. And the TW snippet takes 0.1ms. Plus it reads pretty nice.

Reply:
TIL :)

Reply:
ok, here's what changed...

TW only works on documents, so I move it up to the entire Doc (instead of each template). This now removes every comment, including the license in the html. Should I fix that?

some craziness with tests (jsdom's document doesn't support innerHtml)

treewalker, just like childNodes, doesn't like it when you mutate during iteration. silly web apis don't error when you do that so that was annoying to figure out.